src/Workspaces/CoreTest/UtilityTest/ReferenceHolderTests.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using Roslyn.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests
+{
+    public class ReferenceHolderTests
+    {
+        [Fact]
+        public void SameStrongObjectsEqual()
+        {
+            var obj = new object();
+            var first = ReferenceHolder<object?>.Strong(obj);
+            var second = ReferenceHolder<object?>.Strong(obj);
+
+            VerifyEqual(first, second);
+        }
+
+        [Fact]
+        public void SameWeakObjectsEqual()
+        {
+            var obj = new object();
+            var first = ReferenceHolder<object?>.Weak(obj);
+            var second = ReferenceHolder<object?>.Weak(obj);
+
+            VerifyEqual(first, second);
+        }
+
+        [Fact]
+        public void SameMixedObjectsEqual()
+        {
+            var obj = new object();
+            var first = ReferenceHolder<object?>.Strong(obj);
+            var second = ReferenceHolder<object?>.Weak(obj);
+
+            VerifyEqual(first, second);
+        }
+
+        [Fact]
+        public void NullValuesEqual()
+        {
+            var first = ReferenceHolder<object?>.Strong(null);
+            var second = ReferenceHolder<object?>.Weak(null);
+
+            VerifyEqual(first, second);
+        }
+
+        [Fact]
+        public void ExpiredValueNotEqualToNull()
+        {
+            var strongNull = ReferenceHolder<object?>.Strong(null);
+            var weakNull = ReferenceHolder<object?>.Weak(null);
+            var expired = ReferenceHolder<object?>.TestAccessor.ReleasedWeak(EqualityComparer<object?>.Default.GetHashCode(null));
+
+            Assert.Equal(strongNull.GetHashCode(), expired.GetHashCode());
+            VerifyNotEqual(strongNull, expired);
+            VerifyNotEqual(weakNull, expired);
+        }
+
+        [Fact]
+        public void ExpiredSameValuesEqual()
+        {
+            var first = ReferenceHolder<object?>.TestAccessor.ReleasedWeak(1);
+            var second = ReferenceHolder<object?>.TestAccessor.ReleasedWeak(1);
```

Comment:
Named parameters on these, since I was confused at first why we these were equal since I was forgot the parameter was hash code, not the actual object.

Reply:
➡️ Done